File: java/property-based-testing/src/test/java/org/example/RoverPropertyTest.java
package org.example;


import net.jqwik.api.ForAll;
import net.jqwik.api.Property;

import static org.assertj.core.api.Assertions.assertThat;
import static org.example.Rover.roverBuilder;

public class RoverPropertyTest {

    @Property
    void anyCommand_roverAlwaysChanged(@ForAll Command command) {
        final Rover rover = roverBuilder().build();

        rover.processCommand(command);

        assertThat(rover).isNotEqualTo(roverBuilder().build());
    }
    
}

File: kotlin/property-based-testing/src/test/kotlin/org/example/RoverPropertyTest.kt
package org.example

import net.jqwik.api.ForAll
import net.jqwik.api.Property
import org.assertj.core.api.Assertions.assertThat


class RoverPropertyTest {

    @Property
    fun `anyCommand Rover Always Changed`(@ForAll command: Command) {
        val rover: Rover = aDefaultRover().receiveCommands(command)
        assertThat(rover).isNotEqualTo(aDefaultRover())
    }
}
